### ISSUE TITLE
Can't access columns in a pandas like manner after we sub-select from an initial search result.  

### ISSUE THREAD

**Issue by tylerapritchard** [tylerapritchard/lksearch#14]
We can access a search object pandas DataFrame column initially after a fresh search.  If we sub-select from that, and then try to reference a pandas column we get an empty result returned.  
e.g. 
```
toi = TESSSearch('TOI 1161', sector=14)
toi['pipeline']
```

works as expected

`toi.timeseries['pipeline']`
breaks with returning an empty search result

`toi.timeseries.pipeline` 
works

![image](https://github.com/tylerapritchard/lksearch/assets/28809715/13606edf-f11f-480c-b455-997f0e6b9637)

**Comment by d-giles** [comment]
Can you provide more context for the bug? For python 3.12.3, pandas 2.2.2 both commands work, though they do return different types `toi.timeseries.pipeline` returns a numpy.array) and `toi.timeseries["pipeline"]` returns a pandas.Series.

**Comment by tylerapritchard** [comment]
Hi Dan - I have a minimum working example GIST here: https://gist.github.com/tylerapritchard/c1c56700535ae5c0695b36f2e5ccddb9 - but its pretty similar to the screenshot I included.  

To be clear - the issue I am seeing is that  toi.timeseries["pipeline"] (a TESSSearch Object) is returning a blank/empty MASTSearch object when trying to access its pipeline column.  That column should exist as we see from toi.timeseries, and toi.timeseries.pipeline can still access these values successfully.  

This bug does not occur when I perform this same test on the toi variable itself and not its sub-selected table via the .timeseries property. It does still occur if I set toi.timeseries to a new variable.  

If you are not seeing this behaviour, I wonder if this is an environment issue.  What's your deepcopy version?  That might be the next most likely issue as its used in the _mask function that sub-selects data from the table and returns it as a new TESSSearch object   - I updated my python/pandas to be consistent with yours and I am still seeing this same behavior.
